docs/architecture/adr-038-state-listening.md

```diff
@@ -241,6 +241,131 @@ type StreamingService interface {
     ABCIListener
     // Closer interface
     io.Closer
+=======
+	// ListenBeginBlock updates the streaming service with the latest BeginBlock messages
+	ListenBeginBlock(blockHeight int64, req []byte, res []byte) error
+	// ListenEndBlock updates the steaming service with the latest EndBlock messages
+	ListenEndBlock(blockHeight int64, req []byte, res []byte) error
+	// ListenDeliverTx updates the steaming service with the latest DeliverTx messages
+	ListenDeliverTx(blockHeight int64, req []byte, res []byte) error
+	// ListenStoreKVPair updates the steaming service with the latest StoreKVPair messages
+	ListenStoreKVPair(blockHeight int64, data []byte) error
+}
+
+// StreamingService struct for registering WriteListeners with the BaseApp and updating the service with the ABCI messages using the hooks
+type StreamingService struct {
```

Comment:
This conflicts with the `StreamingService` interface. Consider renaming `StreamingService` interface.

Reply:
Actually, I don't understand the difference between the two...

Reply:
It's due to the merge conflict (which it did not warn me about locally 😞). The interface has been changed to a struct. I'll fix it shortly.